HM3CoreKill/HM3CoreKill/ck/HM3Game.cpp

```diff
@@ -198,6 +202,29 @@ void HM3Game::patchFreeBeamHere()
 	HM3Function::overrideInstruction(HM3_PROCESS_NAME, HM3Offsets::ZHM3CheatMenu_BeamHereFuncPatch, { 0x90, 0x90 });
 }
 
+void HM3Game::setupLoadAnimationHook()
+{
+	/*
+		It's better to relocate some original instructions instead read non original data but for now it doesn't matter for us.
+	*/
+	HM3Function::overrideInstruction(HM3_PROCESS_NAME, 0x00519B0B, { 0x90, 0x90, 0x90 });
+
+	HM3Function::hookFunction<void(__stdcall)(ioi::hm3::ZAnimationInfo*), 5>(
+		HM3_PROCESS_NAME,
+		0x00519B09,
+		(DWORD)ZHM3_OnAnimationLoaded,
+		{
+			x86_pushad,
+			x86_pushfd,
+			x86_push_eax
+		},
+			{
+				x86_popfd,
+				x86_popad,
+				0xC2, 0x04, 0x00 //retn 4
```

Comment:
Possible can be useful to have definition for `x86_ret`, `x86_ret_4`, `x86_ret_8`

Reply:
Something like that
```
#define x86_ret     0xC3
#define x86_ret_4   0xC2, 0x04, 0x00
#define x86_ret_8   0xC2, 0x08, 0x00
```